src/shared/RefreshButton/RefreshButton.tsx

```diff
@@ -13,23 +14,31 @@ export const RefreshButton: VoidFunctionComponent<RefreshButtonProps> = ({
   ariaLabel,
   onClick,
   isDisabled,
+  tooltip,
   isRefreshing,
 }) => {
   const { t } = useTranslation("common");
+
   return (
-    <Button
-      variant="plain"
-      aria-label={ariaLabel || t("refresh_button_label")}
-      isDisabled={isDisabled}
-      onClick={isDisabled === true ? undefined : onClick}
+    <Tooltip
+      content={
+        tooltip
+          ? tooltip
+          : isRefreshing
+          ? t("refreshing_tooltip")
+          : t("refresh_description")
+      }
     >
```

Comment:
@riccardo-forina does this look alright?

Reply:
Yes, but try to avoid nested ternary operators. Something like what I suggested is more readable.

Reply:
I added those suggestions :)